docs/metadata/prophecy-build-tool/pbt-jenkins.md

```diff
@@ -0,0 +1,304 @@
+- Point the "Script path" [to our Jenkinsfile](#deploy-pipeline---pipeline-code) containing the deploy logic
+
+<details>
+<summary> Refer to this image for all settings. </summary>
+
+![hello-cicd-declarative-deploy_configure.png](img%2Fhello-cicd-declarative-deploy_configure.png)
+
+</details>
+
+### Deploy Pipeline - Trigger
+
+Set up a simple webhook trigger for this Job inside of Github.
+
+- Navigate to `Settings > Webhooks > Add Webhook`
+- Create a new webhook like this:
+  ![github-push-webhook-config.png](img%2Fgithub-push-webhook-config.png)
+
+### Deploy Pipeline - Pipeline Code
+
+Create a Groovy Jenkinsfile in the project repository at the below location (relative to root)
+
+```
+.jenkins/test-declarative.groovy
+```
+
+Use the following code as a template, replacing the Fabric IDs and URLs as necessary:
+
+```groovy
+def DEFAULT_FABRIC = "1174"
+def fabricPerBranch = [
+        prod: "4004",
+        qa: "4005",
+        develop: DEFAULT_FABRIC
+]
+
+pipeline {
+    agent any
+    environment {
+        DATABRICKS_HOST =  credentials("${env.GIT_BRANCH == "prod" ? "DEMO_PROD_DATABRICKS_HOST" : "DEMO_DATABRICKS_HOST"}")
+        DATABRICKS_TOKEN = credentials("${env.GIT_BRANCH == "prod" ? "DEMO_PROD_DATABRICKS_TOKEN" : "DEMO_DATABRICKS_TOKEN"}")
+
+        PROJECT_PATH = "./hello_project"
+        VENV_NAME = ".venv"
+        FABRIC_ID = fabricPerBranch.getOrDefault("${env.GIT_BRANCH}", DEFAULT_FABRIC)
+    }
+    stages {
+        stage('prepare system') {
+            steps {
+                sh "apt-get install -y python3-venv"
+            }
+        }
+        stage('install pbt') {
+            steps {
+                sh """#!/bin/bash -xe
+                python3 --version
+                if [[ ! -d "$VENV_NAME" ]]; then
+                    python3 -m venv $VENV_NAME
+                fi
+                source ./$VENV_NAME/bin/activate
+                python3 -m pip install -U pip
+                python3 -m pip install -U build pytest wheel pytest-html pyspark prophecy-build-tool
+                """
+            }
+        }
+        stage('deploy') {
+            steps {
+                sh """
+                . ./$VENV_NAME/bin/activate
+                python3 -m pbt deploy --fabric-ids $FABRIC_ID --path $PROJECT_PATH
+                """
+            }
+        }
+    }
+}
+```
+
+### Deploy Pipeline - Explanation of Pipeline
+
+1. ('environment' block) - Choose which Prophecy Fabric / Databricks Workspace to deploy jobs to.
+
+:::caution
+
+The DATABRICKS_HOST and DATABRICKS_TOKEN env variables must match the configuration of the Fabric we are attempting to deploy.
+
+:::
+
+2. ('prepare system') - Ensure python3 is available on worker nodes. You can skip this if your Jenkins nodes already have `python3-venv` installed.
+3. ('install pbt') - Ensure PBT and its dependencies are installed
+4. ('deploy') - use PBT to deploy the Databricks jobs for our chosen Fabric
```

Comment:
I added some step numbers here, but...steps 5-7 seem incorrect or out of order, right @neontty

Reply:
ugh. the precommit hook kind of garbled this a few times. I'm trying to mix numbered list with sub bullets. I will fix